core-common/src/main/java/eu/arrowhead/common/processor/NetworkAddressDetector.java

```diff
@@ -0,0 +1,212 @@
+package eu.arrowhead.common.processor;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import eu.arrowhead.common.CoreCommonConstants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.common.processor.model.AddressDetectionResult;
+import eu.arrowhead.common.verifier.NetworkAddressVerifier;
+
+@Component
+public class NetworkAddressDetector {
+	
+	//=================================================================================================
+	// methods
+	
+	@Value(CoreCommonConstants.$USE_NETWORK_ADDRESS_DETECTOR_WD)
+	private boolean useDetector;
+	
+	@Value(CoreCommonConstants.$FILTER_PROXY_ADDRESSES_WD)
+	private String[] filterProxyAddressList;
+	private final Set<String> filterProxyAddressSet = new HashSet<>();
+	
+	@Autowired
+	private NetworkAddressPreProcessor networkAddressPreProcessor;
+	
+	@Autowired
+	private NetworkAddressVerifier networkAddressVerifier;
+	
+	private static final String HEADER_FORWARDED = "forwarded";
+	private static final String HEADER_X_FORWARDED_FOR = "x-forwarded-for";
+	
+	private static final String DOT = ".";
+	private static final String COMMA = ",";
+	private static final String COLON = ":";
+	private static final String SEMI_COLON = ";";
+	private static final String DOUBLE_QUOTE = "\"";
+	private static final String SQUARE_BRACKET_OPEN = "[";
+	private static final String SQUARE_BRACKET_CLOSE = "]";
+	private static final String EQUAL_SIGN = "=";
+	private static final int IPV4_DOT_NUMBER = 3;
+	
+	private final Logger logger = LogManager.getLogger(NetworkAddressDetector.class);
+	
+	//-------------------------------------------------------------------------------------------------
+	public AddressDetectionResult detect(final HttpServletRequest servletRequest) { // TODO junit
+		logger.debug("detect started...");
+		final AddressDetectionResult result = new AddressDetectionResult();
+		
+		if (!useDetector || servletRequest == null) {
+			result.setSkipped(true);
+			result.setDetectionMessage("Address-Port detection process was skipped");
+			return result;
+		}
+		boolean detected = retrieveFromConnector(servletRequest, result);
+		
+		if (!detected) {
+			detected = detectByHeaderForwarded(servletRequest, result);			
+		}
+		if (!detected) {
+			detected = detectByHeaderXForwardedFor(servletRequest, result);
+		}
+		
+		if (detected) {
+			try {
+				networkAddressVerifier.verify(result.getDetectedAddress()); // already normalized with NetworkAddressPreProcessor
+				return result;
```

Comment:
"unknown" identifier (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) can slip through I think.

Reply:
Maybe we should put "unkown" in NetworkAddressVerifier as a black-listed expression. Because, the primary  job of this implemetation is to detect the possibile candidates (and pass the to the verifier). What do you think?

Reply:
"unknown" can be a valid hostname, but if it appears in a Forwarded header's "for" subfield, then it have a special meaning. So I put this check into the detector somewhere

Reply:
ok